tsup.config.ts

```diff
@@ -0,0 +1,10 @@
+import { defineConfig } from 'tsup';
+
+export default defineConfig({
+  entry: ['src/index.ts'],
+  dts: true,
+  sourcemap: true,
+  clean: true,
+  format: ['esm', 'cjs', 'iife'],
```

Comment:
The **cjs** build works as expected.

For **iife** the `ethereum-cryptography` dependency has a `@noble/hashes` sub-dependency which requires that the `tsup` `platform` build configuration value to be set to `browser`. It might make sense to remove the `format` configuration property and add corresponding `package.json` scripts to be run in addition to the main `build` before publishing, e.g.
``` 
"build:esm": "tsup --clean false --format esm --platform node",
"build:iife": "tsup --clean false --format iife --platform browser",
...
```
 The **esm** build would necessitate that the imports for several commonjs modules be refactored to work properly: 
- `elliptic` in `src/utils/ellipticCurve.ts`, extract destructuring
- `isomorphic-fetch` in `src/utils/fetchPonyfill.ts`, replace `require`
- `ethereum-cryptography` in `src/utils/hash.ts`, add file extensions
- `bn.js` in `src/utils/number.ts`, extract destructuring